clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -184,7 +184,7 @@ private static long findMinTime(final Collection<? extends RequestState> request
      * completed future if no request is generated.
      */
     public CompletableFuture<Void> maybeAutoCommit(final Map<TopicPartition, OffsetAndMetadata> offsets) {
-        if (!autoCommitState.isPresent()) {
+        if (!canAutoCommit()) {
```

Comment:
pretty terrible words, not sure there's a better description for it.  It needs to check 1. if autocommit is enabled and 2. if there's anything to commit.  If neither, then we don't try to send a commit.

Reply:
maybe `shouldAutoCommit`? Because it's more about being obliged to commit and not about being able to commit? Also, try to be consistent with `AutoCommit` vs. `Autocommit`.

Reply:
In fact, let me change the name of 
```
        public boolean canSendAutocommit() {
            return !this.hasInflightCommit && this.timer.isExpired();
        }
```
to `shouldAutoCommit, because when the timer has expired, we should send the autocommit.